WCFDataService/Client/System/Data/Services/Client/Materialization/ODataEntityMaterializer.cs

```diff
@@ -414,6 +414,22 @@ internal static ODataEntry ProjectionGetEntry(MaterializerEntry entry, string na
             return result.Entry;
         }
 
+        /// <summary>Provides support for getting payload entries or null during projections.</summary>
+        /// <param name="entry">Entry to get sub-entry from.</param>
+        /// <param name="name">Name of sub-entry.</param>
+        /// <returns>The sub-entry or null.</returns>
+        internal static ODataEntry ProjectionGetEntryOrNull(MaterializerEntry entry, string name)
```

Comment:
This function is same with ProjectionGetEntry unless a check to entry access. Consider to extract to a method to reduce duplication.